api/management/commands/migrate_field_report_number.py

```diff
@@ -0,0 +1,64 @@
+import re
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from api.logger import logger
+from api.models import FieldReport
+
+
+class Command(BaseCommand):
+    help = "Migrate legacy FieldReport data to populate fr_num"
+
+    def handle(self, *args, **kwargs):
+        suffix_pattern = re.compile(r"#(\d+)")
+
+        reports = FieldReport.objects.filter(summary__icontains="#", event__isnull=False, countries__isnull=False)
```

Comment:
Let's exclude any event/country field reports with fr_num field with value.

Something like this
```python
# Only process event/country where we don't have fr_num defined
event_country_filter_qs = Report.objects.order_by().values('event', 'countries').annoate(
    count=models.Count('id', filter=models.Q(fr_num__isnull=True))
).exclude(count__gte=1)

report_to_process_qs = Report.objects.filter(
    # Still slow but better then doing this within python
    event__in=event_country_filter_qs.values('event'),
    countries__in=event_country_filter_qs.values('countries'),
).distinct()
```